lib/datasource.js

```diff
@@ -468,23 +468,21 @@ DataSource.prototype.defineRelations = function(modelClass, relations) {
   // Set up the relations
   if (relations) {
     Object.keys(relations).forEach(function(rn) {
+      var targetModel;
       var r = relations[rn];
+
       assert(DataSource.relationTypes.indexOf(r.type) !== -1, 'Invalid relation type: ' + r.type);
       assert(isValidRelationName(rn), 'Invalid relation name: ' + rn);
 
-      var targetModel, polymorphicName;
-
       if (r.polymorphic && r.type !== 'belongsTo' && !r.model) {
-        throw new Error(g.f('No model specified for {{polymorphic}} %s: %s', r.type, rn));
+        throw new Error(g.f('Param "model" not specified for {{polymorphic}} %s relation: %s', r.type, rn));
       }
-
-      if (r.polymorphic) {
-        polymorphicName = typeof r.model === 'string' ? r.model : rn;
```

Comment:
we should not be trying to fallback to relation name to find the polymorphic relation modelTo, it's counter intuitive and misleads people